quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/IgnoredArtifacts.java

```diff
@@ -120,18 +125,34 @@ private static Set<String> fips() {
             .collect(Collectors.toUnmodifiableSet());
 
     private static Set<String> jdbcDrivers() {
-        final Database.Vendor vendor = Configuration.getOptionalValue("quarkus.datasource.db-kind")
-                .flatMap(Database::getVendorByDbKind)
-                .orElse(Database.Vendor.H2);
-
-        final Set<String> jdbcArtifacts = switch (vendor) {
-            case H2 -> JDBC_H2;
-            case MYSQL -> JDBC_MYSQL;
-            case MARIADB -> JDBC_MARIADB;
-            case POSTGRES -> JDBC_POSTGRES;
-            case MSSQL -> JDBC_MSSQL;
-            case ORACLE -> JDBC_ORACLE;
-        };
+        final Converter<String> optionalString = value -> StringUtil.isNotBlank(value) ? value : "";
+
+        final Set<Database.Vendor> vendorsOfAllDatasources = Configuration.getConfig()
+                .getOptionalValues("quarkus.datasource", key -> key, optionalString)
```

Comment:
AFAIK when specifying the key/value types for the `getOptionalValues`, they're handled as required, which means null values are not accepted. However, when some props are empty, their values are managed as null, even when the value type is specified as String. This approach just transforms the null values into empty strings. 

@radcortez JFYI

Reply:
It might be simpler to do this directly:

```
        Set<Database.Vendor> vendorsOfAllDatasources = new HashSet<>();
        Configuration.getConfig().getPropertyNames().forEach(p -> {
            if (p.startsWith("quarkus.datasource.") && p.endsWith(".db-kind")) {
                Configuration.getConfig().getOptionalValue(p, String.class).flatMap(Database::getVendor)
                        .ifPresent(vendorsOfAllDatasources::add);
            }
        });
```

The only other corner case to consider is if someone wants to use a driver dependency, but is not using a quarkus datasource. Should there be any doc note around that, or just wait until it comes up?

Reply:
@shawkins ok, works for me.

> The only other corner case to consider is if someone wants to use a driver dependency, but is not using a quarkus datasource. Should there be any doc note around that, or just wait until it comes up?

I'd consider it a non-standard not supported case, so I would not care much about it ATM, is it ok?

Reply:
Sounds good.

Reply:
@radcortez Thanks for the additional info